### PR TITLE
Add breadcrumbs to my account pages

### LINKED ISSUE #18122: Last breadcrumb element "Your account" is not clickable in the Your account blocks
#### Describe the bug

In the "Your account" blocks, the last breadcrumb element is not clickable.
For example in the  "Your address" page
image

#### Expected behavior

We should have `Home / Your account / Your address`

So, `Your account` is clickable

#### Steps to Reproduce

1. Go to FO 
2. Sign in
3. Go to "Your account" page
4. Check all blocks
5. The last breadcrumb element is not clickable

It is not the same case with PS1.7.6

**Screenshots**
PS1.7.6.4: https://drive.google.com/file/d/1YdUwkeHvEPqe6s4x_41mtf6XoSJR-R6j/view
PS1.7.7: https://drive.google.com/file/d/1aErDxq9Q6c7oX-bCPXOj2Y_VK2nDNvvr/view

#### Additional information

* PrestaShop version: 1.7.7.x
* PHP version: 7.2

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Adds missing breadcrumbs to my acount, login, password and registration pages, to make navigation easier for the customer. I used the same strings from the templates.
| Type?         | improvement
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? |  #16836 and https://github.com/PrestaShop/PrestaShop/issues/18122
| How to test?  | -

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/16837)

### CODE REVIEW

**Comment by PierreRambaud** [comment]
Could you run php-cs-fixer please?

**Comment by Hlavtox** [comment]
@PierreRambaud Done sir, its fine now 🚀

**Review by PierreRambaud — DISMISSED** [review]

**Comment by Hlavtox** [comment]
@PierreRambaud @sarahdib I slightly improved the breadcrumbs. I added order reference as the last key on orderdetail and guesttracking.

Check it again, please.

File: controllers/front/OrderDetailController.php [hunk 1]
```diff
@@ -209,6 +213,11 @@ public function getBreadcrumbLinks()
             'url' => $this->context->link->getPageLink('history'),
         );
 
+        $breadcrumb['links'][] = array(
+            'title' => $this->reference,
```
Comment (PierreRambaud) [inline controllers/front/OrderDetailController.php]: I think you must check if reference is not empty

Comment (Hlavtox) [reply controllers/front/OrderDetailController.php]: Added

**Comment by Hlavtox** [comment]
@PierreRambaud Check it out

**Review by PierreRambaud — DISMISSED** [review]

**Comment by Progi1984** [comment]
@Hlavtox Thanks. Could you fix Pretty CI, please ?

**Comment by Hlavtox** [comment]
@Progi1984 Done boss 🍻

**Review by PierreRambaud — DISMISSED** [review]

File: controllers/front/OrderDetailController.php [hunk 2]
```diff
@@ -209,6 +213,13 @@ public function getBreadcrumbLinks()
             'url' => $this->context->link->getPageLink('history'),
         ];
 
+        if (!empty($this->reference)) {
+            $breadcrumb['links'][] = [
+                'title' => $this->reference,
+                'url' => '#',
```
Comment (Progi1984) [inline controllers/front/OrderDetailController.php:219]: @Hlavtox Is it normal the '#' ?

File: controllers/front/GuestTrackingController.php [hunk 3]
```diff
@@ -154,6 +154,13 @@ public function getBreadcrumbLinks()
             'url' => '#',
         ];
 
+        if (Validate::isLoadedObject($this->order)) {
+            $breadcrumbLinks['links'][] = [
+                'title' => $this->order->reference,
+                'url' => '#',
```
Comment (Progi1984) [inline controllers/front/GuestTrackingController.php:160]: @Hlavtox Is it normal the '#' ?

**Comment by Progi1984** [comment]
Thanks @Hlavtox. Could you rebase your PR to the the `1.7.7.x` branch, please ?

**Comment by Hlavtox** [comment]
@Progi1984 I don't know how to do it, sorry... 🙄  Last time I tried to rebase a PR, it fucked it up completely and added like 300 commits to it.

Update: Merged develop into the branch again, I want to kill myself..........

Should I make a new PR and base the branch on 1.7.7.x?

**Comment by Progi1984** [comment]
Yes, please. Your fix is planified to the milestone 1.7.7.0, so we need it on the branch 1.7.7.x. Thanks :)

**Comment by matks** [comment]
@Hlavtox @Progi1984 rebasing a PR from one branch to another is a complex git operation, please let me do this one as @Hlavtox has contributed multiple times to this repo and deserves help and credit 😉 

@Progi1984 @PrestaShop/prestashop-core-developers Rebasing a PR on its mother branch is something but switching the base branch of a PR is another, a lot harder. I personally do it with cherry-pick for all commits, it's the method that I found is the easiest for me ; however I think it requires advanced git skills.

Shall we improve how we deal with this kind of usecase ? For example we could merge this PR into `develop` then backport it ourselves ? This lets the PR go with the standard flow and get merged, and we take the extra work needed for backport ?

**Comment by matks** [comment]
@Hlavtox Here it is : https://github.com/PrestaShop/PrestaShop/pull/18221/ I backported the 15 commits of logic you added and ignore the merge commits and clean commits for PHP CS Fixer. Can you take a look ? I checked content seems identical to your PR.

@Progi1984 The more I think about it, the more I think it's not a rebase, it's a backport.

**Comment by PululuK** [comment]
> @Hlavtox @Progi1984 rebasing a PR from one branch to another is a complex git operation, please let me do this one as @Hlavtox has contributed multiple times to this repo and deserves help and credit wink
> 
> @Progi1984 @PrestaShop/prestashop-core-developers Rebasing a PR on its mother branch is something but switching the base branch of a PR is another, a lot harder. I personally do it with cherry-pick for all commits, it's the method that I found is the easiest for me ; however I think it requires advanced git skills.
> 
> Shall we improve how we deal with this kind of usecase ? For example we could merge this PR into `develop` then backport it ourselves ? This lets the PR go with the standard flow and get merged, and we take the extra work needed for backport ?


Big up man !! 👏🏾👏🏾👏🏾👏🏾

**Comment by Progi1984** [comment]
May be we can close this PR ? It has been backported to 1.7.7.x branch (and will arrive in develop branch).

**Pull request closed without merging**